<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CommunicationTestForm</class>
 <widget class="QWidget" name="CommunicationTestForm">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>712</width>
    <height>376</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>发送数据测试</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>内容发送</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>数据包发送对象：</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QPushButton" name="pushButton_cameraName">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">border-image: none;
border-color: rgba(255, 255, 255, 0);
background-image:  none;
image: none;
background-color: rgba(255, 255, 255, 0);</string>
         </property>
         <property name="text">
          <string>PushButton</string>
         </property>
         <property name="icon">
          <iconset resource="socketclientuiimage.qrc">
           <normaloff>:/images/IoOn.png</normaloff>:/images/IoOn.png</iconset>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>数据包类型：</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>功能码：</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="3">
        <widget class="QComboBox" name="comboBox_funcCode"/>
       </item>
       <item row="3" column="2" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>3</number>
         </property>
         <item>
          <widget class="QPushButton" name="pushButton_selectImage">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>选择图片</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_imagePath">
           <property name="styleSheet">
            <string notr="true">background-color: rgba(255, 255, 255, 80);</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>内容：</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2" rowspan="2" colspan="3">
        <widget class="QTextEdit" name="textEdit_context"/>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="2">
        <widget class="QPushButton" name="pushButton_send">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>50</height>
          </size>
         </property>
         <property name="text">
          <string>发送</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="6" column="4">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="spacing">
          <number>2</number>
         </property>
         <item row="1" column="0" rowspan="2" colspan="2">
          <widget class="QLabel" name="label_6">
           <property name="maximumSize">
            <size>
             <width>20</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="2">
          <widget class="QSpinBox" name="spinBox_loopTime">
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="suffix">
            <string> ms</string>
           </property>
           <property name="maximum">
            <number>9999999</number>
           </property>
           <property name="value">
            <number>1000</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="3">
          <widget class="QCheckBox" name="checkBox_loop">
           <property name="text">
            <string>循环发送</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="2" colspan="3">
        <widget class="QComboBox" name="comboBox_dataType">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>ack</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>request</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>log</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" colspan="2">
        <widget class="DisplayLogWidget" name="textEdit_log">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>155</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_clearLog">
         <property name="text">
          <string>清除log</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DisplayLogWidget</class>
   <extends>QTextEdit</extends>
   <header>displaylogwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="socketclientuiimage.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>checkBox_loop</sender>
   <signal>toggled(bool)</signal>
   <receiver>pushButton_send</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>318</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_loop</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_funcCode</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_loop</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBox_loopTime</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>379</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
